daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -264,6 +265,111 @@ packagingTests = testGroup "packaging"
+                      (JWT.HMACSecret "secret")
+                      mempty
+                      mempty
+                        { JWT.unregisteredClaims =
+                            JWT.ClaimsMap $
+                            Map.fromList
+                              [ ( "https://daml.com/ledger-api"
+                                , Aeson.Object $
+                                  HashMap.fromList
+                                    [ ("actAs", Aeson.toJSON ["Alice" :: T.Text])
+                                    , ("ledgerId", "MyLedger")
+                                    , ("applicationId", "foobar")
+                                    ])
+                              ]
+                        }
+              let headers = [("Authorization", "Bearer " <> T.encodeUtf8 token)] :: RequestHeaders
+              waitForHttpServer
+                (threadDelay 100000)
+                ("http://localhost:" <> show jsonApiPort <> "/v1/query")
+                headers
+              writeFileUTF8 (projDir </> "daml/Main.daml") $
+                unlines
+                  [ "module Main where"
+                  , "import Daml.Script"
+                  , "template S with newFieldName : Party where signatory newFieldName"
+                  , "init : Script ()"
+                  , "init = do"
+                  , "  [aliceDetails] <- listKnownParties"
+                  , "  let alice = party aliceDetails"
+                  , "  alice `submit` createCmd (S alice)"
+                  , "  pure ()"
+                  ]
+              maybe (fail "No start process stdin handle") (\h -> hPutChar h 'r' >> hFlush h) startStdIn
+              threadDelay 25000000
+              initialRequest <- parseRequest $ "http://localhost:" <> show jsonApiPort <> "/v1/query"
+              let queryRequest =
+                    initialRequest
+                      { method = "POST"
+                      , requestHeaders = headers
+                      , requestBody =
+                          RequestBodyLBS $
+                          Aeson.encode $ Aeson.object ["templateIds" Aeson..= [Aeson.String "Main:S"]]
+                      }
+              manager <- newManager defaultManagerSettings
+              queryResponse <- httpLbs queryRequest manager
+              statusCode (responseStatus queryResponse) @?= 200
+              case Aeson.decode (responseBody queryResponse) of
+                Just (Aeson.Object body)
+                  | Just (Aeson.Array result) <- HashMap.lookup "result" body ->
+                    case Vector.toList result of
+                      [Aeson.Object r]
+                        | Just (Aeson.Object payload) <- HashMap.lookup "payload" r ->
+                          HashMap.lookup "newFieldName" payload @?= Just "Alice"
+                      other -> assertFailure $ "Bad result in response: " <> show other
+                other -> assertFailure $ "Expected JSON object in response body: " <> show other
```

Comment:
Would it simplify things if built an aeson parser here? If so, let's do that in a separate PR.

Reply:
yes, definitely clearer. Let's do a cleanup PR that adds some clarity to the assistant integration tests.